### PR TITLE
AppVeyor: Install stack without using Chocolatey

### PR DESCRIPTION
**Pull request by sjakobi**
Chocolatey is too unreliable.

Fixes https://github.com/dhall-lang/dhall-haskell/issues/1221.

### CODE REVIEW

File: appveyor.yml [hunk 1]
```diff
@@ -19,9 +19,13 @@ environment:
     - STACK_YAML: stack-lts-6.yaml
 
 install:
-  - choco install -y haskell-stack --version %STACK_VERSION%
-  - stack setup > nul
+  # http://help.appveyor.com/discussions/problems/6312-curl-command-not-found
+  - set PATH=C:\Program Files\Git\mingw64\bin;%PATH%
+
+  - curl -sS -ostack.zip -L --insecure "https://github.com/commercialhaskell/stack/releases/download/v%STACK_VERSION%/stack-%STACK_VERSION%-windows-x86_64.zip"
```
Comment (Gabriel439) [inline appveyor.yml]: I try to use long option names for commands that are run non-interactively to better document them:

```suggestion
  - curl --silent --show-error --output stack.zip --location --insecure "https://github.com/commercialhaskell/stack/releases/download/v%STACK_VERSION%/stack-%STACK_VERSION%-windows-x86_64.zip"
```

Also, is the `--insecure` necessary when fetching from `github.com`?  I thought that was only for domains with an invalid certificate

**Comment by sjakobi** [comment]
Merging to put an end to the build cache invalidations. If you have any comments @jneira, just let me know! :)

**Comment by jneira** [comment]
Hey, lgtm to me, but i've found build branches with appveyor in its name useful to make tests about specific issues with it

**Comment by sjakobi** [comment]
> i've found build branches with appveyor in its name useful to make tests about specific issues with it

Ah, now I understand why we had that rule! Would you mind if we bring back that rule as `/.*appveyor-build-branch.*/` or something like that? `/.*appveyor.*/` also covers branch names for changing the AppVeyor config and similar, where the build times are already way too long due to the cache invalidation.

**Comment by jneira** [comment]
Yeah, something shorter would be great but that one could work too

**Pull request merged**